pad.go

```diff
@@ -185,9 +277,163 @@ func handleViewMeta(app *App, w http.ResponseWriter, r *http.Request) error {
 		return ErrPostNotFound
 	}
 	appData.Flashes, _ = getSessionFlashes(app, w, r, nil)
+	user := getUserSession(app, r)
+
+	if slug == "" {
+		slug, _ = getSlugFromActionId(app, action)
+	}
 
+	mediaDirectoryPath := filepath.Join(app.cfg.Server.MediaParentDir, mediaDir,
+						user.Username, slug)
+	appData.Post.MediaFilesList, _ = getFilesListInPath(mediaDirectoryPath)
 	if err = templates["edit-meta"].ExecuteTemplate(w, "edit-meta", appData); err != nil {
 		log.Error("Unable to execute template: %v", err)
 	}
 	return nil
 }
+
+func getNewFileName(path string, originalFieName string) (string, error) {
+	u, err := uuid.NewV4()
+	if err != nil {
+		log.Error("Unable to generate uuid: %v", err)
+		return "", err
+	}
+	extension := filepath.Ext(originalFieName)
+	return u.String() + extension, nil
+}
+
+func getFilesListInPath(path string) ([]string, error) {
+	var files []string
+	entries, err := ioutil.ReadDir(path)
+	if err != nil {
+		return nil, err
+	}
+	for _, entry := range entries {
+		if !entry.IsDir() {
+			files = append(files, "/" + path + "/" + entry.Name())
+		}
+	}
+	return files, nil
+}
+
+func handleGetFile(app *App, w http.ResponseWriter, r *http.Request) error {
+	vars := mux.Vars(r)
+	slug := vars["slug"]
+	author := vars["author"]
+	filename := vars["filename"]
+	mediaDirectoryPath := filepath.Join(app.cfg.Server.MediaParentDir, mediaDir,
+							                                   author, slug, filename)
+	filePath := mediaDirectoryPath 
+	file, err := http.Dir("").Open(filePath)
+	if err != nil {
+		http.Error(w, "File not found", http.StatusNotFound)
+		return nil
+	}
+	defer file.Close()
+	fileInfo, err := file.Stat()
+	if err != nil {
+		http.Error(w, "Failed to get file information", http.StatusInternalServerError)
+		return nil
+	}
+	w.Header().Set("Content-Disposition", "attachment; filename="+fileInfo.Name())
+	w.Header().Set("Content-Type", "application/octet-stream")
```

Comment:
Files should probably be delivered as their correct MIME type, instead of `application/octet-stream`. There are default Go packages that help with this.